web/server/api/endpoints/plan.py

```diff
@@ -130,28 +139,61 @@ def get_plan(
     return plan
 
 
-def _get_plan_changes(plan: Plan) -> t.Dict[str, t.Any]:
+def _get_plan_changes(context: Context, plan: Plan) -> models.PlanChanges:
     """Get plan changes"""
-    return {
-        "removed": set(plan.context_diff.removed_snapshots),
-        "added": list(plan.context_diff.added),
-        "modified": models.ModelsDiff.get_modified_snapshots(plan.context_diff),
-    }
+    snapshots = plan.context_diff.snapshots
+    default_catalog = context.default_catalog
+
+    def _display_name(snapshot_id: SnapshotId) -> str:
```

Comment:
Nit: Can we DRY this up? I've noticed this (and `_node_type()`) copy and pasted in several places. Maybe it's not worth doing since you can't make use of closure.

Reply:
I added 
```
@staticmethod
def get_view_name
```
to `ChangeDisplay` class to use instead of `_display_name`